R/dataframe.R

```diff
@@ -182,24 +182,53 @@ as_data_frame.matrix <- function(x, ...) {
   as_data_frame(as.data.frame(x))
 }
 
-#' Convert row names to an explicit variable.
+#' Converting between rownames and a column
+#'
+#' add_rownames: Convert row names to an explicit variable.
 #'
 #' @param df Input data frame with rownames.
 #' @param var Name of variable to use
 #' @export
+#' @rdname add_rownames
 #' @examples
 #' tbl_df(mtcars)
 #'
-#' add_rownames(mtcars)
+#' mtcars_tbl <- add_rownames(mtcars)
 add_rownames <- function(df, var = "rowname") {
   stopifnot(is.data.frame(df))
 
+  if (var %in% colnames(df) ) {
+    stop(paste("There is a column named", var, "already!"))
+  }
+
   rn <- as_data_frame(setNames(list(rownames(df)), var))
-  rownames(df) <- NULL
+  rownames(rn) <-rn[[var]]
```

Comment:
Why would you keep the row names of rn and df? The code looked good to me before. (R has "pass by value" semantics, which means that logically df is a copy of the original data frame and can be modified at will. Unless it's a data table...)